Update project level docs (#1734)
paritytech/parity-bridges-common: docs/high-level-overview.md

source chain and target chain nodes. Relayers are reading state of the source chain, compare it to the
state of the target chain and, if state at target chain needs to be updated, submits target chain
transaction.

### GRANDPA Finality Relay

The task of relay is to submit source chain GRANDPA justifications and their corresponding headers to
the Bridge GRANDPA Finality Pallet, deployed at the target chain. For that, the relay subscribes to
the GRANDPA justifications stream and submits every new justification it sees. In addition, relay is
searching for mandatory headers and submits their justifications - without that the pallet will be
unable to move forward.

More: [GRANDPA Finality Relay Sequence Diagram](./grandpa-finality-relay.html), [code](../relays/finality/).

### Parachains Finality Relay

The relay connects to the source _relay_ chain and the target chain nodes. It doesn't need to connect to the
tracked parachain nodes. The relay looks at the [`Heads`](https://github.com/paritytech/polkadot/blob/1a034bd6de0e76721d19aed02a538bcef0787260/runtime/parachains/src/paras/mod.rs#L642)

> map of the [`paras` pallet](https://github.com/paritytech/polkadot/tree/1a034bd6de0e76721d19aed02a538bcef0787260/runtime/parachains/src/paras) and compares the value with the best parachain head, stored at the bridge parachains pallet at

Reviewer: ```suggestion
map of the [`paras` pallet](https://github.com/paritytech/polkadot/tree/1a034bd6de0e76721d19aed02a538bcef0787260/runtime/parachains/src/paras)
in sourche chain, and compares the value with the best parachain head, stored in the bridge parachains pallet at
```

> the taget chain. If new parachain head appears at the relay chain block `B`, the relay process **waits**
> until header `B` or one of its ancestors appears at the target chain. Once it is available, the storage
> proof of the map entry is generated and is submitted to the target chain.

Reviewer: I think it waits for `B` or _children_ of `B` (block numbers > B), rather than _ancestors_ (parents) of `B`... right?

```suggestion
the taget chain.
When new parachain head appears at source relay chain block `B`, the relay process **waits**
until header `B` or one of its children blocks' finality is relayed to and appears on target chain.
Once it is available, the storage proof of the map entry is generated and is submitted to the
target chain.
```

As its on-chain component (which requires bridge GRANDPA pallet to be deployed nearby), the parachains
finality relay requires GRANDPA finality relay to be running in parallel. Without it, the header `B` or

> some its ancestor won't be finalized and we'll be unable to verify generated storage proof. 

Author: Are you sure? I mean, `some of` is for selecting several items from the set - e.g. "some of its ancestors"may mean "ancestor `B-1` and ancestor `B-2`" and here I mean one of, so "some its ancestor" (like `B-1`). Sorry - probably I'm dumb here, just wanted to clarify :)

> some its ancestor won't be finalized and we'll be unable to verify generated storage proof. 

Reviewer: If I understand correctly, the least ambiguous option would be `one of its ancestors` or simply `some ancestor`.

> some its ancestor won't be finalized and we'll be unable to verify generated storage proof. 

Reviewer: ```suggestion
any of its children's finality at source won't be relayed at target, and target chain
won't be able to verify generated storage proof. 
```

More: [Parachains Finality Relay Sequence Diagram](./parachains-finality-relay.html), [code](../relays/parachains/).

### Messages Relay

> Messages relay is actually a two relays that are running in a single process: messages delivery relay and

Reviewer: ```suggestion
Messages relay is actually two relays that are running in a single process: messages delivery relay and
```

delivery confirmation relay. Even though they are more complex and have many caveats, the overall algorithm
is the same as in other relays.

Message delivery relay connects to the source chain and looks at the outbound lane end, waiting until new
messages are queued there. Once they appear at the source block `B`, the relay start waiting for the block

> `B` or its ancestor appear at the target chain. Then the messages storage proof is generated and submitted

Reviewer: again: afaik in blockchain terms, `ancestor` means `(grand(grand..))parent` - maybe `children` or `descendants`?
```suggestion
`B` or its descendant appear at the target chain. Then the messages storage proof is generated and submitted
```

to the bridge messages pallet at the target chain. In addition, the transaction may include the storage proof
of the outbound lane state - that proves that relayer rewards have been paid and this data (map of relay
accounts to the delivered messages) may be pruned from the inbound lane state at the target chain.

Delivery confirmation relay connects to the target chain and starts watching the inbound lane end. When new
messages are delivered to the target chain, the corresponding _source chain account_ is inserted to the
map in the inbound lane data. Relay detects that, say, at the target chain block `B` and waits until that

> block or its ancestor appears at the source chain. Once that happens, the relay crafts a storage proof of

Reviewer: ```suggestion
block or its descendant appears at the source chain. Once that happens, the relay crafts a storage proof of
```
